homeassistant/components/device_tracker/config_entry.py

```diff
@@ -82,27 +82,27 @@ def location_accuracy(self):
         return 0
 
     @property
-    def location_name(self) -> str:
+    def location_name(self) -> str | None:
         """Return a location name for the current location of the device."""
         return None
 
     @property
-    def latitude(self) -> float:
+    def latitude(self) -> float | None:
         """Return latitude value of the device."""
-        return NotImplementedError
+        raise NotImplementedError
 
     @property
-    def longitude(self) -> float:
+    def longitude(self) -> float | None:
         """Return longitude value of the device."""
-        return NotImplementedError
+        raise NotImplementedError
 
     @property
     def state(self):
         """Return the state of the device."""
         if self.location_name:
             return self.location_name
 
-        if self.latitude is not None:
+        if self.latitude is not None and self.longitude is not None:
```

Comment:
It's because of this line. I don't see any reason to treat latitude and longitude differently.

Right now if only latitude is set, state is `not_home`, but if only longitude is set, state is `unknown`. I assume the test was written to match current behaviour.

Reply:
We assume both are set of one is set.